ci-operator/step-registry/ipi/conf/gcp/public-dns-zone/ipi-conf-gcp-public-dns-zone-commands.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+if ! test -f "${SHARED_DIR}/xpn.json"; then
+  echo "'${SHARED_DIR}/xpn.json' not found, abort." && exit 1
+fi
+
+GCP_BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+
+echo "Reading variables from ${SHARED_DIR}/xpn.json..."
+HOST_PROJECT=$(jq -r '.hostProject' "${SHARED_DIR}/xpn.json")
+
+if [[ -z "${BASE_DOMAIN}" ]]; then
+  BASE_DOMAIN="$(jq -r '.baseDomain' "${SHARED_DIR}/xpn.json")"
+fi
+if [[ -z "${BASE_DOMAIN_ZONE_NAME}" ]]; then
+  BASE_DOMAIN_ZONE_NAME=$(jq -r '.baseDomainZoneName' "${SHARED_DIR}/xpn.json")
+fi
+if [[ -z "${BASE_DOMAIN_ZONE_PROJECT}" ]]; then
+  BASE_DOMAIN_ZONE_PROJECT="${HOST_PROJECT}"
+fi
```

Comment:
When users are using public domain from host project, `BASE_DOMAIN` should be always be overridden accordingly, no need the above if` conditions, right?

Reply:
An alternative is, specifying them in the workflow or the chain.

Reply:
A principal rule in prow ci is opt-in needed steps in workflow stead of judging ENV vars, so in this case, I think we only take getting domain info from host project into consideration.